src/frequenz/sdk/timeseries/battery_pool/_battery_pool.py

```diff
@@ -36,143 +33,132 @@
+        self._priority = priority
 
-        self._namespace: str = f"battery-pool-{self._batteries}-{uuid.uuid4()}"
-        self._power_distributing_namespace: str = f"power-distributor-{self._namespace}"
-        self._channel_registry: ChannelRegistry = channel_registry
-        self._formula_pool: FormulaEnginePool = FormulaEnginePool(
-            self._namespace,
-            channel_registry,
-            resampler_subscription_sender,
-        )
-
-    async def set_power(
+    async def propose_power(
         self,
-        power: Power,
+        power: Power | None,
         *,
-        adjust_power: bool = True,
         request_timeout: timedelta = timedelta(seconds=5.0),
         include_broken_batteries: bool = False,
+        bounds: timeseries.Bounds[Power | None] = timeseries.Bounds(None, None),
     ) -> None:
-        """Set the given power for the batteries in the pool.
+        """Send a proposal to the power manager for the pool's set of batteries.
 
         Power values need to follow the Passive Sign Convention (PSC). That is, positive
         values indicate charge power and negative values indicate discharge power.
 
-        When not using the Passive Sign Convention, the `charge` and `discharge` methods
-        might be more convenient.
+        If the same batteries are shared by multiple actors, the power manager will
+        consider the priority of the actors, the bounds they set, and their preferred
+        power, when calculating the target power for the batteries.
 
-        The result of the request can be accessed using the receiver returned from
-        the `power_distribution_results` method.
+        The preferred power of lower priority actors will take precedence as long as
+        they respect the bounds set by higher priority actors.  If lower priority actors
+        request power values outside of the bounds set by higher priority actors, the
+        target power will be the closest value to the preferred power that is within the
+        bounds.
+
+        When there are no other actors trying to use the same batteries, the actor's
+        preferred power would be set as the target power, as long as it falls within the
+        system power bounds for the batteries.
+
+        The result of the request can be accessed using the receiver returned from the
+        `power_status` method, which also streams the bounds that an actor should
+        comply with, based on its priority.
 
         Args:
-            power: The power to set for the batteries in the pool.
-            adjust_power: If True, the power will be adjusted to fit the power bounds,
-                if necessary. If False, then power requests outside the bounds will be
-                rejected.
+            power: The power to propose for the batteries in the pool.  If None, the
+                proposed power of higher priority actors will take precedence as the
+                target power.
```

Comment:
I find the comment about `None` a bit confusing. Isn't the intent of passing `None` to sort of *withdraw* a proposal, like not participating in the decision of how much power to set? If so I would mention something more explicit along these lines to make more clear what's the use case, not only what is the effect.

Also nitpick:
```suggestion
            power: The power to propose for the batteries in the pool.  If `None`, the
                proposed power of higher priority actors will take precedence as the
                target power.
```

Reply:
It is a bit tricky, because there could be bounds even if power is `None`.  If both of them are `None`, that's like withdrawing the proposal,  else it is just "not having a perferred power".

Reply:
OK, I see, to be honest I didn't read all the docs, so if this is well explained somewhere, we should be fine. It is complicated so we can't have a full explanation everywhere, agreed.

Reply:
maybe this looks better?

```suggestion
            power: The power to propose for the batteries in the pool.  If None, this
                proposal will not have any effect on the target power, unless bounds
                are specified.  If both are `None`, it is equivalent to not having a 
                proposal or withdrawing a previous one.
```

Reply:
Yup!

Reply:
done in: https://github.com/frequenz-floss/frequenz-sdk-python/commit/79ecc0d2